backend/app/auth/service.py

```
@@ -282,27 +251,19 @@ async def authenticate_with_google(self, id_token: str) -> Dict[str, Any]:
                 )
 
             return {"user": user, "refresh_token": refresh_token}
-        except HTTPException:
-            raise
+
+        except firebase_auth.InvalidIdTokenError:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Invalid Google ID token",
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)